lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -363,27 +365,245 @@
     platform.registerFunction(platform.FunctionKind.Array_forEach, function (callbackfn, thisArg = undefined) {
         // ECMAScript 2017 #sec-array.prototype.foreach
 
+        //Let O be ? ToObject(this value).
+        //Let len be ? ToLength(? Get(O, "length")).
         let objInfo = __chakraLibrary.CheckArrayAndGetLen(this, "Array.prototype.forEach");
         let o = objInfo.o;
         let len = objInfo.len;
-
+        
+        //If IsCallable(callbackfn) is false, throw a TypeError exception.
         if (typeof callbackfn !== "function") {
             __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.forEach");
         }
 
+        //If thisArg is present, let T be thisArg; else let T be undefined.
+        //Let k be 0.
         let k = 0;
 
+        //Repeat, while k < len
         while (k < len) {
+            //Let Pk be ! ToString(k).
+            //Let kPresent be ? HasProperty(O, Pk).
+            //If kPresent is true, then
             if (k in o) {
+                //Let kValue be ? Get(O, Pk).
                 let kValue = o[k];
+                //Perform ? Call(callbackfn, T, « kValue, k, O »).
                 __chakraLibrary.builtInCallInstanceFunction(callbackfn, thisArg, kValue, k, o);
             }
+            //Increase k by 1.
             k++;
         }
-
+        //Return undefined. 
         return undefined;
     });
 
+    platform.registerFunction(platform.FunctionKind.Array_some, function (callbackfn, thisArg = undefined) {
```

Comment:
spoken in person, but please check the typedarray performance change on these apis.